Shopify/hydrogen
# Fix pagination internal state

### WHY are these changes introduced?

Fixes https://github.com/Shopify/hydrogen/issues/1290





### WHAT is this pull request doing?
1. Change the Pagination component to reset internal state when the URL changes (not including Pagination params).
2. Validate `pageInfo` within the `connection` prop

### HOW to test your changes?

1. Boot up the demo store products page: http://localhost:3000/products or use https://jpwynq314-f1f4fa724b7467f41f07.myshopify.dev/
2. Open up the console, make sure there aren't hydration errors.
3. Click load more. More products should load.
4. Refresh and make sure no hydration errors.
5. Copy the link and open in an incognito window. A previous link should appear. Click it, previous products should load and the link should disappear. Refresh the page. No hydration errors should occur.
6. Click load more, and more products should append.
7. Select a filter, the pagination state should reset and the filter apply.

#### Post-merge steps



#### Checklist
